### PR TITLE
add aws-deploy workflow

### PR DESCRIPTION
**Pull request by djeung**
- add `aws-deploy.yml` github workflow for adding to `public_data_releases` 
- update existing workflow to `node v18`
- update existing actions to github commit hashes for security

**Pull request recommendations:**

- [ ] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [ ] Link to any relevant issue in the PR description. Ex: _Resolves [gh-##], adds tiff file format support_
- [ ] Provide description and context of changes.
- [ ] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

File: .github/workflows/aws-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+          # Compute a short sha for use in the OIDC session name, which has a 64 character limit
+          - name: Add SHORT_SHA env property with commit short sha
+            run: echo "SHORT_SHA=`echo ${{ github.sha }} | cut -c1-8`" >> $GITHUB_ENV
+
+          - name: Configure AWS credentials with OIDC
+            uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502
+            with:
+                role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/github_nucmorph_colorizer
+                role-session-name: nucmorph_colorizer-${{ env.SHORT_SHA }}
```
Comment (toloudis) [inline .github/workflows/aws-deploy.yml:66]: should the role be named volumeviewer or is this reusing a pre-existing role?

File: .github/workflows/aws-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      steps:
+        - name: Checkout
+          uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+  
+        - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
+          with:
+            node-version: "18"
+  
+        - name: Install and Build
+          run: npm ci && npx vite build
```
Comment (toloudis) [inline .github/workflows/aws-deploy.yml:41]:
```suggestion
          run: npm ci && npm run gh-build
```
(Minor difference from nucmorph colorizer's build command)

File: .github/workflows/aws-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  
+        - name: Upload build files
+          uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
+          with:
+            name: aws-deploy-files
+            path: ./dist
```
Comment (toloudis) [inline .github/workflows/aws-deploy.yml:47]:
```suggestion
            path: ./imageviewer
```

**Review by toloudis — COMMENTED** [review]
One question, plus a couple of suggestions to get us closer to a working build for this repo...

File: .github/workflows/aws-deploy.yml (hunk 1 above)
Comment (djeung) [reply .github/workflows/aws-deploy.yml:66]: ah, bad copy/paste

**Review by toloudis — APPROVED** [review]
LGTM and thanks for updating the other workflows

File: .github/workflows/aws-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+          - name: Configure AWS credentials with OIDC
+            uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502
+            with:
+                role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/github_website_3d_cell_viewer
+                role-session-name: website_3d_cell_viewer-${{ env.SHORT_SHA }}
+                aws-region: ${{ env.AWS_REGION }}
+
+          # Setup variables based on the staging or production environment
+          - name: Set ECS variables based on environment
```
Comment (anl) [inline .github/workflows/aws-deploy.yml:70]: Nit: Not an ECS deployment
```suggestion
          - name: Set deployment variables based on environment
```

**Review by anl — APPROVED** [review]

**Review by griffinfujioka — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -75,18 +75,20 @@ jobs:
...
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+      - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
         with:
           node-version: ${{env.NODE_VERSION}}
           registry-url: https://registry.npmjs.org/
       - run: npm ci
       - run: npm run gh-build
       - name: Deploy
-        uses: peaceiris/actions-gh-pages@v3
+        uses: peaceiris/actions-gh-pages@4eb285e828117bca26638192c3ed309c622e7bad
```
Comment (toloudis) [inline .github/workflows/publish.yml:86]: apparently this commit is not a release and doesn't work at all.  Maybe try the latest tagged release

File: .github/workflows/nightly.yml [hunk 6]
```diff
@@ -37,7 +37,7 @@ jobs:
           echo "::set-output name=tag_name::${TAG_NAME}"
           echo "::set-output name=deploy_tag_name::deploy-${TAG_NAME}"
       - name: Deploy
-        uses: peaceiris/actions-gh-pages@v3
+        uses: peaceiris/actions-gh-pages@4eb285e828117bca26638192c3ed309c622e7bad
```
Comment (toloudis) [inline .github/workflows/nightly.yml:40]: apparently this commit is not a release and doesn't work at all. Maybe try the latest tagged release

Comment (anl) [reply .github/workflows/nightly.yml:40]: FWIW, I use this Bash function to find the SHA(s) of an Action's tag:

```
get_sha_for_action() {
    # Split the input into repo and tag
    repo="${1%@*}"        # Extracts everything before the '@'
    tag="${1#*@}"         # Extracts everything after the '@'

    # Fetch the tag information using curl and the GitHub API
    tag_info=$(curl -s -H "Accept: application/vnd.github.v3+json" \
    "https://api.github.com/repos/$repo/git/refs/tags/$tag")

    # Check if the tag is an annotated tag
    if echo "$tag_info" | grep -q '"type": "tag"'; then
        # Annotated tag
        tag_sha=$(echo "$tag_info" | jq -r '.object.sha')
        commit_sha=$(curl -s -H "Accept: application/vnd.github.v3+json" \
        "https://api.github.com/repos/$repo/git/tags/$tag_sha" | jq -r '.object.sha')

        echo "Annotated tag '$tag' found."
        echo "Tag SHA: $tag_sha"
        echo "Commit SHA: $commit_sha"
    else
        # Lightweight tag
        commit_sha=$(echo "$tag_info" | jq -r '.object.sha')

        echo "Lightweight tag '$tag' found."
        echo "Commit SHA: $commit_sha"
    fi
}
```

**Pull request merged**